### PR TITLE
RDM-4073 add utility methods to CaseField 

### PR DESCRIPTION
**Pull request by tzarsmango**
https://tools.hmcts.net/jira/browse/RDM-4073

Added utility methods `isComplex` and `isCollection` to CaseField. This required many changes to the test code due to the method used for CaseField creation in tests.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/shared/domain/definition/case-field.model.ts [hunk 1]
```diff
@@ -25,4 +25,12 @@ export class CaseField implements Orderable {
...
   @Type(() => WizardPageField)
   wizardProps?: WizardPageField;
+
+  isCollection(): boolean {
+    return this.field_type && this.field_type.type === 'Collection';
+  }
+
+  isComplex(): boolean {
+    return this.field_type && this.field_type.type === 'Complex';
+  }
```
Comment (mario-paniccia) [inline src/shared/domain/definition/case-field.model.ts]: @tzarsmango I know is not part of the ticket, but I was thinking whether we could also add other methods like isLabel() (I think to remember that's another one very often used:     return field.field_type.type === 'Label';
) and some others you can think of
Feel free to ignore

File: src/shared/fixture/case-field-builder.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import { CaseField, FieldType } from '../domain/definition';
+import { AccessControlList } from '../domain/definition/access-control-list.model';
+
+export class CaseFieldBuilder {
```
Comment (andrewfolga) [inline src/shared/fixture/case-field-builder.ts]: 👍

File: src/shared/fixture/case-field.test.fixture.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { FieldType } from '../domain/definition';
+import { textFieldType } from '.';
+import { CaseFieldBuilder } from './case-field-builder';
+
+export let newCaseField = (id: string,
```
Comment (andrewfolga) [inline src/shared/fixture/case-field.test.fixture.ts]: 👍

File: src/shared/domain/definition/case-field.model.ts (hunk 1 above)
Comment (mario-paniccia) [reply src/shared/domain/definition/case-field.model.ts]: and a method
Also a method:
getComplexSubfield(path) that allows retrieval of a Complex field subfield by its path.
This logic currently sits in  wizard-page-field-to-case-field.mapper.ts and could be moved in CaseField.

@tzarsmango if you decide to not tackle these as part of this story, please let me know I will create a separate ticket

**Pull request closed without merging**